dune_client/api/execution.py

```diff
@@ -66,26 +71,90 @@ def get_execution_status(self, job_id: str) -> ExecutionStatusResponse:
         except KeyError as err:
             raise DuneError(response_json, "ExecutionStatusResponse", err) from err
 
-    def get_execution_results(self, job_id: str) -> ResultsResponse:
+    def get_execution_results(
+        self,
+        job_id: str,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+    ) -> ResultsResponse:
         """GET results from Dune API for `job_id` (aka `execution_id`)"""
-        response_json = self._get(route=f"/execution/{job_id}/results")
+        params = {}
+        if limit is not None:
+            params["limit"] = limit
+        if offset is not None:
+            params["offset"] = offset
+
+        route = f"/execution/{job_id}/results"
+        url = self._route_url(route)
+        return self.get_execution_results_by_url(url=url, params=params)
+
+    def get_execution_results_by_url(
+        self,
+        url: str,
+        params: Optional[Dict[str, Any]] = None,
+    ) -> ResultsResponse:
+        """
+        GET results from Dune API with a given URL. This is particularly useful for pagination.
+
+        No checks are performed on the URL; it is your responsibility to ensure
+        it is a legitimate URL, or one returned directly by the API.
+        The API key will be sent to the server.
+        """
+        response_json = self._get(url=url, params=params)
         try:
             return ResultsResponse.from_dict(response_json)
         except KeyError as err:
             raise DuneError(response_json, "ResultsResponse", err) from err
 
-    def get_execution_results_csv(self, job_id: str) -> ExecutionResultCSV:
+    def get_execution_results_csv(
+        self,
+        job_id: str,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+    ) -> ExecutionResultCSV:
         """
         GET results in CSV format from Dune API for `job_id` (aka `execution_id`)
 
         this API only returns the raw data in CSV format, it is faster & lighterweight
         use this method for large results where you want lower CPU and memory overhead
         if you need metadata information use get_results() or get_status()
         """
+        params = {}
+        if limit is not None:
+            params["limit"] = limit
+        if offset is not None:
+            params["offset"] = offset
+
         route = f"/execution/{job_id}/results/csv"
-        response = self._get(route=route, raw=True)
+        url = self._route_url(route)
+        return self.get_execution_results_csv_by_url(url=url, params=params)
+
+    def get_execution_results_csv_by_url(
```

Comment:
Should these by_url methods be private?

Reply:
I don't feel super strongly about it, but thought that it might clutter the "auto-complete" in people's IDEs. Maybe its better to add an optional `endpoint_url` to the `get_execution_results` method... and make this private (triggered only when the optional parameter is not None). We may also want to test for DuneErrors arising when a user specifies a bogus url.